codec/src/main/java/io/netty/handler/codec/compression/Lz4FrameDecoder.java

```diff
@@ -212,69 +213,48 @@ protected void decode(ChannelHandlerContext ctx, ByteBuf in, List<Object> out) t
-                                throw new DecompressionException(String.format(
-                                    "stream corrupted: compressedLength(%d) and actual length(%d) mismatch",
-                                    compressedLength, readBytes));
-                            }
-                        } catch (LZ4Exception e) {
-                            throw new DecompressionException(e);
+                            decompressor.decompress(inNioBuffer(in),
+                                    uncompressed.internalNioBuffer(uncompressed.writerIndex(), decompressedLength));
+                            // Update the writerIndex now to reflect what we decompressed.
+                            uncompressed.writerIndex(uncompressed.writerIndex() + decompressedLength);
+                            break;
                         }
-                        break;
-                    }
-                    default:
-                        throw new DecompressionException(String.format(
-                                "unexpected blockType: %d (expected: %d or %d)",
-                                blockType, BLOCK_TYPE_NON_COMPRESSED, BLOCK_TYPE_COMPRESSED));
+                        default:
+                            throw new DecompressionException(String.format(
+                                    "unexpected blockType: %d (expected: %d or %d)",
+                                    blockType, BLOCK_TYPE_NON_COMPRESSED, BLOCK_TYPE_COMPRESSED));
                     }
+                    // Skip inbound bytes after we processed them.
+                    in.skipBytes(compressedLength);
 
-                    final Checksum checksum = this.checksum;
                     if (checksum != null) {
-                        checksum.reset();
-                        checksum.update(dest, destOff, decompressedLength);
-                        final int checksumResult = (int) checksum.getValue();
-                        if (checksumResult != currentChecksum) {
-                            throw new DecompressionException(String.format(
-                                    "stream corrupted: mismatching checksum: %d (expected: %d)",
-                                    checksumResult, currentChecksum));
-                        }
+                        CompressionUtil.checkChecksum(checksum, uncompressed, currentChecksum);
                     }
-                    uncompressed.writerIndex(uncompressed.writerIndex() + decompressedLength);
                     out.add(uncompressed);
-                    in.skipBytes(compressedLength);
-
+                    uncompressed = null;
                     currentState = State.INIT_BLOCK;
-                    success = true;
+                } catch (LZ4Exception e) {
+                    throw new DecompressionException(e);
                 } finally {
-                    if (!success) {
+                    if (uncompressed != null) {
```

Comment:
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) Change this condition so that it does not always evaluate to "false" [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS2583)